From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>
Bug: https://github.com/gentoo-haskell/gentoo-haskell/issues/1231

doctest pulls in the out-of-scope package `hashing` if it is installed on the
system. A workaround is to use PackageImports to force importing from `cryptonite`.

diff -urN dhall-1.40.1-new1/ghc-src/Dhall/Crypto.hs dhall-1.40.1-new2/ghc-src/Dhall/Crypto.hs
--- dhall-1.40.1-new1/ghc-src/Dhall/Crypto.hs	2001-09-08 19:46:40.000000000 -0600
+++ dhall-1.40.1-new2/ghc-src/Dhall/Crypto.hs	2021-10-31 02:10:15.618404840 -0600
@@ -1,5 +1,6 @@
 {-# LANGUAGE DeriveGeneric              #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE PackageImports             #-}
 
 {-| This module provides implementations of cryptographic utilities that only
     work for GHC (as opposed to GHCJS)
@@ -13,13 +14,13 @@
     ) where
 
 import Control.DeepSeq         (NFData)
-import Crypto.Hash             (SHA256)
+import "cryptonite" Crypto.Hash (SHA256)
 import Data.ByteArray          (ByteArrayAccess, convert)
 import Data.ByteArray.Encoding (Base (Base16), convertToBase)
 import Data.ByteString         (ByteString)
 import GHC.Generics            (Generic)
 
-import qualified Crypto.Hash
+import qualified "cryptonite" Crypto.Hash
 import qualified Data.ByteString.Char8 as ByteString.Char8
 
 -- | A SHA256 digest
